inbc-program/inbc/parser/source_app_parser.py

```diff
@@ -19,15 +27,16 @@ def application_add(args: argparse.Namespace) -> str:
     manifest = ('<?xml version="1.0" encoding="utf-8"?>' +
                 '<manifest>' +
                 '<type>source</type>' +
-                '<applicationSource>' +
-                '<add><gpg>'
-                '{0}' +
-                '{1}'
-                '</gpg><repo><repos>').format(create_xml_tag(arguments, "uri"),
-                                              create_xml_tag(arguments, "keyname"))
+                '<applicationSource>' + '<add>')
+
+    if args.gpgKeyUri and args.gpgKeyName:
+        manifest += ('<gpg>' + '{0}' + '{1}' + '</gpg>').format(create_xml_tag(arguments, "uri"),
+                                                                create_xml_tag(arguments, "keyname"))
+
+    manifest += '<repo><repos>'
 
     for source in args.sources:
-        manifest += '<source_pkg>' + source.strip() + '</source_pkg>'
+        manifest += '<source_pkg>' + source + '</source_pkg>'
```

Comment:
**Suggestion:** Use a list and str.join to build the XML string instead of concatenating within a loop for better performance. [performance]
**NOTE:** Ensure that the suggested code is comprehensive. This serves as an example of the code you could use. The user should verify it after accepting the suggestion.
```suggestion
    source_tags = [f'<source_pkg>{source}</source_pkg>' for source in args.sources]
    manifest += ''.join(source_tags)
```

Reply:
done